src/test/java/gameframework/drawing/GameCanvasDefaultImplTest.java

```diff
@@ -0,0 +1,44 @@
+package gameframework.drawing;
+
+import static org.junit.Assert.*;
+
+import java.awt.event.KeyListener;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import gameframework.drawing.mocks.KeyListenerMock;
+
+public class GameCanvasDefaultImplTest {
+
+	protected GameCanvasDefaultImpl gameCanvas;
+	protected KeyListenerMock keyListener;
+	
+	@Before
+	public void init() {
+		gameCanvas  = new GameCanvasDefaultImpl();
+		keyListener = new KeyListenerMock();
+	}
+	
+	public KeyListener getLatestKeyListenerAdded() {
+		KeyListener[] listeners = gameCanvas.getKeyListeners();
+		return listeners.length > 0 ? listeners[listeners.length - 1] : null;
+	}
+	
+	@Test
+	public void testAddKeyListener() {
+		gameCanvas.addKeyListener(keyListener);
+		assertSame(keyListener, getLatestKeyListenerAdded());
+	}
+
+	@Test
+	public void testRemoveKeyListener() {
+		int expected = gameCanvas.getKeyListeners().length;
+		
+		gameCanvas.addKeyListener(keyListener);		
+		gameCanvas.removeKeyListener(keyListener);
```

Comment:
please assert that the length increased by 1 between the add and remove. That way, you indicate to the future developer that this is `remove` which changes the length.